src/linear_algebra/determinant.lean

```diff
@@ -381,4 +382,77 @@ begin
       rw [hx, from_blocks_apply₂₁], refl }}
 end
 
+/-- Laplacian expansion of the determinant of an `n+1 × n+1` matrix along column 0. -/
+lemma det_succ_column_zero {n : ℕ} (A : matrix (fin n.succ) (fin n.succ) R) :
+  det A = ∑ i : fin n.succ, (-1) ^ (i : ℕ) * A i 0 *
+    det (A.minor i.succ_above fin.succ) :=
+begin
+  rw [matrix.det_apply, finset.univ_perm_fin_succ, ← finset.univ_product_univ],
+  simp only [finset.sum_map, equiv.to_embedding_apply, finset.sum_product, matrix.minor],
+  refine finset.sum_congr rfl (λ i _, fin.cases _ (λ i, _) i),
+  { simp only [fin.prod_univ_succ, matrix.det_apply, finset.mul_sum,
+        equiv.perm.decompose_fin_symm_apply_zero, fin.coe_zero, one_mul,
+        equiv.perm.decompose_fin.symm_sign, equiv.swap_self, if_true, id.def, eq_self_iff_true,
+        equiv.perm.decompose_fin_symm_apply_succ, fin.succ_above_zero, equiv.coe_refl, pow_zero,
+        algebra.mul_smul_comm] },
+  -- `univ_perm_fin_succ` gives a different embedding of `perm (fin n)` into
+  -- `perm (fin n.succ)` than the determinant of the submatrix we want,
+  -- permute `A` so that we get the correct one.
+  have : (-1 : R) ^ (i : ℕ) = i.cycle_range.sign,
+  { simp [fin.sign_cycle_range] },
+  rw [fin.coe_succ, pow_succ, this, mul_assoc, mul_assoc, mul_left_comm ↑(equiv.perm.sign _),
+      ← det_permute, matrix.det_apply, finset.mul_sum, finset.mul_sum],
+  -- now we just need to move the crresponding parts to the same place
+  refine finset.sum_congr rfl (λ σ _, _),
+  rw [equiv.perm.decompose_fin.symm_sign, if_neg (fin.succ_ne_zero i)],
+  calc ((-1) * σ.sign : ℤ) • ∏ i', A (equiv.perm.decompose_fin.symm (fin.succ i, σ) i') i'
+      = ((-1) * σ.sign : ℤ) • (A (fin.succ i) 0 *
+        ∏ i', A (((fin.succ i).succ_above) (fin.cycle_range i (σ i'))) i'.succ) : _
+  ... = (-1) * (A (fin.succ i) 0 * (σ.sign : ℤ) •
+        ∏ i', A (((fin.succ i).succ_above) (fin.cycle_range i (σ i'))) i'.succ) : by simp,
+  simp only [fin.prod_univ_succ, fin.succ_above_cycle_range,
+      equiv.perm.decompose_fin_symm_apply_zero, equiv.perm.decompose_fin_symm_apply_succ]
+end
+
+/-- Laplacian expansion of the determinant of an `n+1 × n+1` matrix along row 0. -/
+lemma det_succ_row_zero {n : ℕ} (A : matrix (fin n.succ) (fin n.succ) R) :
+  det A = ∑ j : fin n.succ, (-1) ^ (j : ℕ) * A 0 j *
+    det (A.minor fin.succ j.succ_above) :=
+by { rw [← det_transpose A, det_succ_column_zero],
+     refine finset.sum_congr rfl (λ i _, _),
+     rw [← det_transpose],
+     simp only [matrix.transpose_apply],
+     refl }
```

Comment:
Can these `refl`s be replaced with rewrites once my PR adding stuff about `minor` is in?